x/gov/client/cli/tx.go

```diff
@@ -430,7 +436,12 @@ $ %s tx gov generate-constitution-amendment path/to/updated/constitution.md
 		},
 	}
 
-	flags.AddTxFlagsToCmd(cmd)
+	// This is not a tx command (but a utility for the proposal tx), so we don't need to add tx flags.
+	// It might actually be confusing, so we just add the query flags.
+	flags.AddQueryFlagsToCmd(cmd)
```

Comment:
`NewCmdDraftProposal()` is also part of the `tx` command and just output some data, similarly. But it doesn't replace the tx flags with the query flags. Why do you need to query flags ?

Reply:
`NewCmdDraftProposal` is a little different also because it's generating a tx right? while this command doesn't, it generates the content for a tx NOT the whole tx so stuff like chain-id or fees doesn't even make sense.
While it could make sense in that case.

But, it's true that adding the tx flags to that command is alwo wrong because there are some flags that let imagine that the command is able to submit the tx too

Reply:
After further checks it seems actually the txflags are added but ignored in the case of `NewCmdDraftProposal`? So shouldn't they be removed altogether? no flags are used aside from the skip-metadata one (generally the flags are used in `tx.GenerateOrBroadcastTxCLI()` but that is not called in the command as the command writes directly to file, actually the name of the file can't even be changed...)

Reply:
Confirmed, so I actually removed it in this PR

Reply:
got it, thanks for the investigation and the report!